plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3359,6 +3365,32 @@ public boolean delegateMaterializedViewRefreshToConnector(ConnectorSession sessi
         return hiveMaterializedViewMetadata.refreshMaterializedView(session, name);
     }
 
+    @Override
+    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        requireNonNull(session, "session is null");
+        requireNonNull(tableName, "tableName is null");
+        if (isHiveSystemSchema(tableName.getSchemaName())) {
+            return Optional.empty();
+        }
+        // we need to chop off any "$partitions" and similar suffixes from table name while querying the metastore for the Table object
+        SchemaTableName tableNameBase = getTableNameBase(tableName);
+        Optional<Table> table = metastore.getTable(new HiveIdentity(session), tableNameBase.getSchemaName(), tableNameBase.getTableName());
+        if (table.isEmpty() || VIRTUAL_VIEW.name().equals(table.get().getTableType())) {
+            return Optional.empty();
+        }
+
+        return tableRedirectionsProvider.redirectTable(session, tableName, table.get());
```

Comment:
With tableName vs tableNameBase being handled here, we should pass only the tableNameBase to tableRedirectionsProvider and re-add chopped off ourselves.